### PR TITLE
Don't send keyboard events without focus

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: src/server/frontend_wayland/wl_keyboard.cpp [hunk 1]
```diff
@@ -49,10 +49,10 @@ void mf::WlKeyboard::handle_event(MirInputEvent const* event, WlSurface& surface
     if (!focused_surface.is(surface))
     {
         log_warning(
```
Comment (Saviq) [inline src/server/frontend_wayland/wl_keyboard.cpp:51]: Shouldn't this be a `log_error` now? It happening would be an error in our WM logic, no?

File: src/server/frontend_wayland/wl_keyboard.cpp [hunk 2]
```diff
@@ -49,10 +49,10 @@ void mf::WlKeyboard::handle_event(MirInputEvent const* event, WlSurface& surface
...
-            "Sending keyboard event to wl_surface@%u even though it was not explicitly given keyboard focus",
+            "Attempting to sending keyboard event to wl_surface@%u even though it was not given keyboard focus",
```
Comment (Saviq) [inline src/server/frontend_wayland/wl_keyboard.cpp:52]:
```suggestion
            "Attempt to send keyboard event to wl_surface@%u even though it was not given keyboard focus",
```

**Review by Saviq — CHANGES_REQUESTED** [review]

File: src/server/frontend_wayland/wl_keyboard.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply src/server/frontend_wayland/wl_keyboard.cpp:51]: Yes, but it should be `fatal_error()`

**Review by Saviq — APPROVED** [review]
Good!

bors merge

**Pull request merged**